src/backend/distributed/commands/multi_copy.c

```diff
@@ -465,7 +465,7 @@ CopyToExistingShards(CopyStmt *copyStatement, char *completionTag)
 	 * PG_CATCH. Otherwise, it may be undefined in the PG_CATCH (see sigsetjmp
 	 * documentation).
 	 */
-	shardConnectionHash = CreateShardConnectionHash();
+	copyConnectionHash = CreateShardConnectionHash(MessageContext);
```

Comment:
One message can trigger a ton of stuff, e.g. a million COPY commands. We'd have to wait for all of that to complete before resetting the memory context. Best not to use MessageContext for anything not directly related to the message itself.